test/helpers/test-with-server.js

```diff
@@ -0,0 +1,22 @@
+const tape = require('tape');
+const createServer = require('../../server');
+const config = require('../../config');
+
+module.exports = (description, cb) => {
+  createServer(config, (err, server) => {
+    if (err) {
+      throw err;
+    }
+
+    server.start(() => {
+      if (err) {
+        throw err;
+      }
+
+      tape(description, (t) => {
+        cb(t, server);
+        server.stop();
```

Comment:
For async operations this will probably stop the server before we've finished testing it.

Since we're probably going to be using `server.inject` in our tests we don't actually need to "start" the server.